server/app/controllers/NewTask.scala

```diff
@@ -39,13 +43,16 @@ class NewTask @Inject()(executor: DynamicSuiteExecutor, dao: Dao, val messagesAp
       f => {
         val futureResponse = for {
           _ <- Future(StringBuilderRunner(executor(f.referenceSolution, f.suite))).check
-          db <- dao.addTask(Task(scalaClass, f.taskDescription, f.solutionTemplate, f.referenceSolution, f.suite))
+          db <- dao.addTask(
+            Task(new Timestamp(System.currentTimeMillis),
+              scalaClass, UUIDs.timeBased(), f.name, f.description, f.solutionTemplate, f.referenceSolution, f.suite)
+          )
         } yield Redirect(routes.Application.index)
 
         futureResponse.recover {
           case NonFatal(e) => Logger.warn(e.getMessage, e)
             BadRequest {
-              views.html.addTask(addTaskForm.bindFromRequest().withError(taskDescription, messagesApi(cannotAddTask)))
+              views.html.addTask(addTaskForm.bindFromRequest().withError(taskDescription, s"${messagesApi(cannotAddTask)}: ${e.getMessage}"))
```

Comment:
It is not going to be our exception, but exception in the reference solution and test suite. Both are provided by user.